dev/PushNotifications/PushNotificationsLongRunningTask/platform.cpp

```diff
@@ -0,0 +1,130 @@
+﻿#include "pch.h"
+
+#include <PushNotificationsLRP_h.h>
+
+#include "platform.h"
+
+wil::unique_event g_winmainEvent(wil::EventOptions::None);
+
+Microsoft::WRL::ComPtr<WpnLrpPlatformImpl> g_platform;
+
+std::once_flag m_isPlatformInitialized;
+
+WpnLrpPlatformImpl* GetPlatform()
+{
+    // We have to call InitializePlatform() before retrieving the platform on the factory.
+    // Throw if Platform is already reset or was shut down.
+    THROW_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, !g_platform || g_platform->IsPlatformShutdown());
```

Comment:
No locks in this method and method below?

Reply:
These are global functions, the locks are eventually called in the member functions.